app/dao/notifications_dao.py

```diff
@@ -70,6 +71,37 @@ def dao_get_last_template_usage(template_id, template_type, service_id):
     ).first()
 
 
+@statsd(namespace="dao")
+def dao_get_last_date_template_was_used(template_id, template_type, service_id):
```

Comment:
do we need this template_type filter here? it seems redundant, and we've got indexes on key_type, service_id and template_id already so i'm not sure it adds any speed-up.